### PR TITLE
[CWG motion 7 2024-06] P0963R3 Structured binding declaration as a condition

### LINKED ISSUE #7083: [2024-06 CWG Motion 7] P0963R3 Structured binding declaration as a condition
https://wiki.edg.com/pub/Wg21stlouis2024/StrawPolls/p0963r3.html

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #7083 
Fixes cplusplus/papers#1617

### CODE REVIEW

File: source/statements.tex [hunk 1]
```diff
@@ -34,6 +34,7 @@
 \nontermdef{condition}\br
     expression\br
     \opt{attribute-specifier-seq} decl-specifier-seq declarator brace-or-equal-initializer
+    structured-binding-declaration initializer
```
Comment (burblebee) [inline source/statements.tex:37]: missing` \br` on line before

File: source/statements.tex [hunk 2]
```diff
@@ -86,18 +87,29 @@
...
+is a declaration\iref{dcl.dcl}.
 The \grammarterm{declarator} shall not
 specify a function or an array. The \grammarterm{decl-specifier-seq} shall not
 define a class or enumeration. If the \keyword{auto} \grammarterm{type-specifier} appears in
 the \grammarterm{decl-specifier-seq},
 the type of the identifier being declared is deduced from the initializer as described in~\ref{dcl.spec.auto}.
 
 \pnum
-The value of a \grammarterm{condition} that is an initialized declaration
+The \defn{decision variable} of a \grammarterm{condition}
```
Comment (burblebee) [inline source/statements.tex:103]: also add an index under variable, like `\indextext{variable!decision variable}`, or use `\defnadj` instead

File: source/statements.tex [hunk 3]
```diff
@@ -86,18 +87,29 @@
...
+that is neither an \grammarterm{expression} nor a structured binding declaration
+is the declared variable.
+The decision variable of a \grammarterm{condition}
+that is a structured binding declaration is specified in \ref{dcl.struct.bind}.
+
+\pnum
+The value of a \grammarterm{condition} that not an \grammarterm{expression}
```
Comment (burblebee) [inline source/statements.tex:110]: "that not" -> "that is not"

File: source/declarations.tex [hunk 4]
```diff
@@ -7042,6 +7047,10 @@
 \tcode{E} is never a reference type\iref{expr.prop}.
 \end{note}
 
+\pnum
+If a structured binding declaration appears as a \grammarterm{condition},
+the decision variable\iref{stmt.pre} of the condition is \tcode{e}.
```
Comment (burblebee) [inline source/declarations.tex:7052]: `\tcode{e}` -> `\exposid{e}`

File: source/declarations.tex [hunk 5]
```diff
@@ -7105,6 +7114,11 @@
 Each $\tcode{v}_i$ is the name of an lvalue of type $\tcode{T}_i$
 that refers to the object bound to $\tcode{r}_i$;
 the referenced type is $\tcode{T}_i$.
+The initialization of \tcode{e} and
+any conversion of \tcode{e} considered as a decision variable\iref{stmt.pre}
```
Comment (burblebee) [inline source/declarations.tex:7118]: \tcode{e} -> \exposid{e}

**Review by burblebee — CHANGES_REQUESTED** [review]

File: source/statements.tex (hunk 1 above)
Comment (jensmaurer) [reply source/statements.tex]: fixed

File: source/statements.tex (hunk 2 above)
Comment (jensmaurer) [reply source/statements.tex]: fixed

File: source/statements.tex (hunk 3 above)
Comment (jensmaurer) [reply source/statements.tex]: fixed

File: source/declarations.tex (hunk 4 above)
Comment (jensmaurer) [reply source/declarations.tex]: fixed

File: source/declarations.tex (hunk 5 above)
Comment (jensmaurer) [reply source/declarations.tex]: fixed

Comment (burblebee) [reply source/declarations.tex]: looks like you only fixed the one on line 7118 - please fix the one on line 7117 also

File: source/declarations.tex [hunk 6]
```diff
@@ -7115,7 +7115,7 @@
 that refers to the object bound to $\tcode{r}_i$;
 the referenced type is $\tcode{T}_i$.
 The initialization of \tcode{e} and
```
Comment (burblebee) [inline source/declarations.tex:7117]: missed this one: \tcode{e} -> \exposid{e}

**Review by burblebee — CHANGES_REQUESTED** [review]

File: source/declarations.tex (hunk 6 above)
Comment (jensmaurer) [reply source/declarations.tex]: fixed

**Review by burblebee — APPROVED** [review]

**Pull request merged**